// select2/Gruntfile.js

const config = {
    cwd: __dirname,
    sass: {
        dist: {
            files: [
                {
                    expand: true,
                    cwd: 'src/scss',
                    src: ['core.scss'],
                    dest: '../../../vendor/bower-asset/bootstrap/src/css',
                    ext: '.css',
                    rename: function (src, dest) {
                        return `${src}/${dest.replace(/core/, 'select2')}`
                    }
                }
            ]
        }
    },
    copy: {
        main: {
            files: [
                {
                    expand: true,
                    cwd: 'dist/js',
                    src: ['select2.full.js'],
                    dest: '../../../vendor/bower-asset/bootstrap/src/js',
                    rename: function (src, dest) {
                        return `${src}/${dest.replace(/\.full/, '')}`
                    }
                },
            ],
        },
    },

    concat: {
        options: {
            separator: ';',
        },
        dist: {
            src: ['dist/js/select2.full.js'],
            dest: '../../../vendor/bower-asset/bootstrap/src/js/select2.js',
        },
    },
}


/** @param {import("grunt")} grunt */
module.exports.Select2Grunt = function (grunt) {
    return grunt.initConfig(config)
}
